engine/src/main/java/org/camunda/bpm/engine/impl/cfg/ProcessEngineConfigurationImpl.java

```diff
@@ -1383,6 +1382,55 @@ public void initDatabaseType() {
     }
   }
 
+  /**
+   * Try to read schema name use jdbc.Make it more self config.
+   * <p>
+   * If we use a oracle database and not set the {@link #databaseSchema}.
+   * The engine {@link #databaseSchemaUpdate} check call
+   * {@link org.camunda.bpm.engine.impl.db.sql.DbSqlSession#isTablePresent(String)}
+   * would read all the schema tables in the database.
+   * <p>
+   * And it would return true if there is another schema have the same name of check table
+   * which will fail the {#databaseSchemaUpdate}
+   */
+  protected void initDatabaseSchema() {
```

Comment:
This code is only intended for oracle databases. Since we support multiple database types we can not accept this solution.

Reply:
Sorry  for that, I wrote a test again and there was no problem .
But I actually got an error when I set the 
> camunda.bpm.database.schema-name=TEST
> camunda.bpm.database.table-prefix=TEST.
with Oracle dababase.

This is because oracle jdbc connection have no setSchema method which will cause database init fail.

Sorry for that again.